src/main/java/org/logstash/dissect/fields/FieldFactory.java

```diff
@@ -0,0 +1,37 @@
+package org.logstash.dissect.fields;
+
+public final class FieldFactory {
+    private static final String MIXED_PREFIXES = "Field cannot prefix with both Append and Indirect Prefix (%s): %s";
+    private static final String PREFIXED_EMPTY = "Field cannot be a prefix on its own without further text";
+
+    public static Field create(String field) {
+        if (field.isEmpty() || field.startsWith("?")) {
```

Comment:
Not having more context: I think it is confusing if the grammar is even spread across the Ruby / Java code base...  or did I misunderstand you?